src/ModelContextProtocol/Utils/Json/McpJsonUtilities.cs

```diff
@@ -74,7 +74,18 @@ private static JsonSerializerOptions CreateDefaultOptions()
     internal static JsonTypeInfo<T> GetTypeInfo<T>(this JsonSerializerOptions options) =>
         (JsonTypeInfo<T>)options.GetTypeInfo(typeof(T));
 
+    internal static void ValidateSerializerOptions(JsonSerializerOptions options)
+    {
+        if (options.WriteIndented)
+        {
+            throw new InvalidOperationException("JsonSerializerOptions.WriteIndented is not supported for JSON-RPC workloads.");
+        }
```

Comment:
Tabs or newlines are invalid?

Reply:
Actually, it turns out that this enforcement is redundant since user-provided JSOs are only used to write parameters into DOM types for which indentation rules do not apply. I will remove this check.